python/xir/ir.lark

```diff
@@ -69,10 +68,10 @@ FALSE_: "false"
 // declarations
 ?declarations: (gate_decl | operator_decl | func_decl | output_decl)
 
-gate_decl: _GATE name "," uint "," uint ";"
-operator_decl: _OP name "," uint "," uint ";"
-func_decl: _FUNC name "," uint ";"
-output_decl: _OUT name ";"
+gate_decl: _GATE name params? wires ";"
```

Comment:
Would it suffice to replace `params?` with `params_list?` everywhere except for the `gatestmt` rule?  Here are some tests:
```python
# Fail because (a: 9) cannot be used to define gate parameters.
gate rA9(a: 9): RX(a) | [0]; end;

# Fail because (a: 9) cannot be used to declare gate parameters.
gate rA9(a: 9);

```